.travis.yml

```diff
@@ -15,28 +15,34 @@ notifications:
 
 matrix:
   include:
-    - os: linux  # list of Linux env: https://docs.travis-ci.com/user/reference/overview/
-      name: "Ubuntu 18.04 (Bionic Beaver)"
-      dist: bionic
-    - os: linux
-      name: "Ubuntu 16.04 (Xenial)"
-      dist: xenial
-    - os: linux
-      name: "Ubuntu 14.04 (Trusty)"
-      dist: trusty
-    - os: osx  # list of OSX env: https://docs.travis-ci.com/user/reference/osx/
-      name: "OSX 10.14 (Mojave)"
-      osx_image: xcode11.2
-    - os: osx
-      name: "OSX 10.13 (High Sierra)"
-      osx_image: xcode9.4
-    - os: osx
-      name: "OSX 10.12 (Sierra)"
-      osx_image: xcode9.2
+    # - os: linux  # list of Linux env: https://docs.travis-ci.com/user/reference/overview/
+    #   name: "Ubuntu 18.04 (Bionic Beaver)"
+    #   dist: bionic
+    # - os: linux
+    #   name: "Ubuntu 16.04 (Xenial)"
+    #   dist: xenial
+    # - os: linux
+    #   name: "Ubuntu 14.04 (Trusty)"
+    #   dist: trusty
+    # - os: osx  # list of OSX env: https://docs.travis-ci.com/user/reference/osx/
+    #   name: "OSX 10.14 (Mojave)"
+    #   osx_image: xcode11.2
+    # - os: osx
+    #   name: "OSX 10.13 (High Sierra)"
+    #   osx_image: xcode9.4
+    # - os: osx
+    #   name: "OSX 10.12 (Sierra)"
+    #   osx_image: xcode9.2
+    - os: windows
+      language: bash
 
 install:
   - |
     echo Installing SCT
+    if [ "${TRAVIS_OS_NAME}" = "windows" ]; then
+      choco install wsl-ubuntu-1804
+      sudo /c/ProgramData/chocolatey/lib/wsl-ubuntu-1804/tools/unzipped/ubuntu1804.exe
+    fi      
```

Comment:
This is much like the issue I had in #2690: you need to launch a sub-environment and continue the script from there.

I solved it by moving the entire Travis script into a separate file and then running that instead: https://github.com/neuropoly/spinalcordtoolbox/blob/4225bd6652707c077e4af2c44678c53d862c6fbf/.travis.sh

It does the important reboot-into-a-different-OS step with:

https://github.com/neuropoly/spinalcordtoolbox/blob/4225bd6652707c077e4af2c44678c53d862c6fbf/.travis.sh#L26-L27

Reply:
I suppose for integrating both WSL-containers and Docker-containers we need the test script to not have that header that assumes docker. Maybe we want this:

```
     if [ "${TRAVIS_OS_NAME}" = "windows" ]; then 
         ./wsl.sh .travis.sh
     elif [ -n "$DOCKER_IMAGE" ]; then
        ./docker.sh .travis.sh
     else
        .travis.sh
      fi
```

With 

```
   # docker.sh
   docker run \
     --name container \
     --init \
     -it -d \
     --rm \
     -v `pwd`:/repo -w /repo \
     "$DOCKER_IMAGE"
   trap "docker stop container" EXIT
   # set up groups to match the owner of /repo, so the installer can't tell
   docker exec container groupadd -g "`id -g`" `id -g -n`
   docker exec container useradd -m -u "`id -u`" -g "`id -g`" `id -u -n`
   docker exec container $DOCKER_DEPS    # install platform-specific dependencies
   # recurse to run the real test script
   # --user `id -u` makes sure the build script is the owner of the files at /repo
   docker exec --user `id -u`:`id -g` container "$1"
   exit 0
```

and

```
      # wsl.sh
      choco install wsl-ubuntu-1804
      sudo /c/ProgramData/chocolatey/lib/wsl-ubuntu-1804/tools/unzipped/ubuntu1804.exe "$1"
```